OpenSim/Simulation/Model/Muscle.cpp

```diff
@@ -144,6 +157,7 @@ void Muscle::constructProperties()
     // By default the min and max controls on muscle are 0.0 and 1.0
     setMinControl(0.0);
     setMaxControl(1.0);
+    upd_GeometryPath().setDefaultColor(DefaultMuscleColor);
```

Comment:
OK I think you are right Ayman. The  Ligament is always green because `extendFinalizeFromProperties` is setting the color to the static const value. I think your constructProperties() intercept is correct with the consumer of the `GeometryPath` having a chance to have its say for the default color as a property. The overwriting of the color in `extendFinalizeFromProperties` should be removed however so that whatever user set value of the property is used.